src/pages/waiting-room/cat-home-test/__tests__/waiting-room.cat-d.page.spec.ts

```diff
@@ -0,0 +1,291 @@
+
+import { AppModule } from '../../../../app/app.module';
+import { WaitingRoomCatHomeTestPage } from '../waiting-room.cat-home-test.page';
+import { AuthenticationProvider } from '../../../../providers/authentication/authentication';
+import { AuthenticationProviderMock } from '../../../../providers/authentication/__mocks__/authentication.mock';
+import { Store, StoreModule } from '@ngrx/store';
+import { StoreModel } from '../../../../shared/models/store.model';
+import {
+  ToggleResidencyDeclaration,
+  ToggleInsuranceDeclaration,
+} from '../../../../modules/tests/pre-test-declarations/common/pre-test-declarations.actions';
+import {
+  initialState as preTestDeclarationInitialState,
+} from '../../../../modules/tests/pre-test-declarations/common/pre-test-declarations.reducer';
+import { DeviceAuthenticationProvider } from '../../../../providers/device-authentication/device-authentication';
+import {
+  DeviceAuthenticationProviderMock,
+} from '../../../../providers/device-authentication/__mocks__/device-authentication.mock';
+import { DateTimeProvider } from '../../../../providers/date-time/date-time';
+import { DateTimeProviderMock } from '../../../../providers/date-time/__mocks__/date-time.mock';
+import { WaitingRoomValidationError } from '../../waiting-room.actions';
+import { TranslateModule, TranslateService } from 'ng2-translate';
+import { Subscription } from 'rxjs/Subscription';
+import * as communicationPreferenceActions
+  from '../../../../modules/tests/communication-preferences/communication-preferences.actions';
+import { Language } from '../../../../modules/tests/communication-preferences/communication-preferences.model';
+import { ScreenOrientation } from '@ionic-native/screen-orientation';
+import { ScreenOrientationMock } from '../../../../shared/mocks/screen-orientation.mock';
+import { Insomnia } from '@ionic-native/insomnia';
+import { InsomniaMock } from '../../../../shared/mocks/insomnia.mock';
+import { MockComponent } from 'ng-mocks';
+import { ConductedLanguageComponent } from '../../components/conducted-language/conducted-language';
+import { InsuranceDeclarationComponent } from '../../components/insurance-declaration/insurance-declaration';
+import { ResidencyDeclarationComponent } from '../../components/residency-declaration/residency-declaration';
+import { SignatureComponent } from '../../components/signature/signature';
+import { EndTestLinkComponent } from '../../../../components/common/end-test-link/end-test-link';
+import { LockScreenIndicator } from '../../../../components/common/screen-lock-indicator/lock-screen-indicator';
+import { CandidateSectionComponent } from '../../../../components/common/candidate-section/candidate-section';
+import { FormControl, Validators } from '@angular/forms';
+import { candidateMock } from '../../../../modules/tests/__mocks__/tests.mock';
+import { CatDUniqueTypes } from '@dvsa/mes-test-schema/categories/D';
```

Comment:
Needs a to-do

Reply:
instead of a TODO - it now uses the right type